build_dependency_tree.py

```diff
@@ -3,76 +3,202 @@
+    return: returns:    a list of the files that `file` depends on
     """
-    with open(file) as f:
-        # some files may have macros that check for operatoring system or 
+    # For some reason does not drop the <> or " when matching
+    # so we write a helper
+    def extract_header_file_name(text: Optional[str]):
+        if text:
+            return text[1:len(text)-1]
+        else:
+            return None
+
+    with open(str(file.absolute())) as f:
+        # some files may have macros that check for operatoring system or
         # compiler compatibilities. For now we ignore these as well.
-        # We also need to ignore include statements made in comments 
+        # We also need to ignore include statements made in comments
         include_statements = []
-        b_is_block_comment = False # Lines that start with /*
+        b_is_block_comment = False  # Lines that start with /*
         for line in f.readlines():
+            print(f"{line.strip()}")
             if b_is_block_comment:
-                if line.split()[0] == '*/':
+                if '*/' in line:
                     b_is_block_comment = False
                 else:
                     continue
             else:
-                if line.contains('/*'):
+                if '/*' in line:
                     b_is_block_comment = True
                     continue
-                elif line.split() == '//':
+                elif line.replace(' ', '')[0:2] == '//':
+                    continue
+                elif not line:  # empty line
                     continue
-                elif line.split()[0] == '#include':
-                    # match either the pattern `<...>` or `"..."`
-                    # re.findall will return a list of tuples `[(pattern 1, pattern 2), ...]`
-                    re_match = re.findall(r"<(.+?.)>|\"(.+?.)\"", line.split()[1])[0]
+                elif line.replace(' ', '')[0:8] == '#include' or \
+                        line.replace(' ', '')[0:9] == '%:include':
+                    # we always expect the included file to be given second
+                    # (required by standards)
+                    re_match = include_statement_pattern.match(line.split()[1])
                     # We only expect one entry in the list
-                    if not re_match[0]: # first matching group is empty
+                    if not re_match:  # first matching group is empty
+                        print(f'No header files found in line: {line}')
                         continue
+                    elif not re_match.group(1):
+                        # Check if match with `<...>` is None
+                        print(f"""Header file found
+                                {str(file.absolute())}:{re_match.group(2)}""")
+                        include_statements.append(
+                                extract_header_file_name(
+                                    re_match.group(2)))
                     else:
```

Comment:
```suggestion
                    if re_match:  # Match found in string literal
                        print(f"""Header file found
                                {str(file.absolute())}:{re_match.group(1)}""")
                        include_statements.append(
                                extract_header_file_name(
                                    re_match.group(1)))
                    else:
```

Reply:
I want to keep both groups for the future.